tests/Common/SpanChecker.php

```diff
@@ -416,18 +418,29 @@ function ($key) use ($pattern) {
                 $namePrefix . "Unexpected extra values for 'tags':\n" . print_r($filtered, true)
             );
             foreach ($exp->getExistingTagNames(isset($span['parent_id'])) as $tagName) {
-                TestCase::assertArrayHasKey($tagName, $spanMeta);
+                if ($tagName === Tag::PID) {
+                    TestCase::assertArrayHasKey($tagName, $spanMetrics);
+                    unset($spanMetrics[Tag::PID]);
+                } else {
+                    TestCase::assertArrayHasKey($tagName, $spanMeta);
+                }
             }
         }
         $metrics = $exp->getExactMetrics();
         if ($metrics !== SpanAssertion::NOT_TESTED) {
             // Ignore compilation-time metric unless explicitly tested
             if (!isset($metrics['php.compilation.total_time_ms'])) {
-                unset($span['metrics']['php.compilation.total_time_ms']);
+                unset($spanMetrics['php.compilation.total_time_ms']);
+            }
+            if (isset($metrics['process_id'])) {
```

Comment:
Are these supposed to be negated with `!` as well? Meaning, unless they are specifically tested, we shouldn't test them?

Reply:
The purpose of these lines is to unset the process_id field in both our expected and actual metric arrays. Doing this because we already asserted on the existence of the process_id field on lines 421-425. If we don't unset the field, we will have a failure on line 441 when comparing exact arrays equal since the process_id values will not be equal.